src/fu-agent.c

```diff
@@ -0,0 +1,645 @@
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_INVALID_ARGS,
+				     "Invalid arguments");
+		return FALSE;
+	}
+
+	/* set up networking */
+	priv->soup_session = fu_util_setup_networking (error);
+	if (priv->soup_session == NULL)
+		return FALSE;
+
+	/* just proxy */
+	return fu_util_agent_sync (priv, error);
+}
+
+static void
+fu_util_ignore_cb (const gchar *log_domain, GLogLevelFlags log_level,
+		   const gchar *message, gpointer user_data)
+{
+}
+
+static gboolean
+fu_util_sigint_cb (gpointer user_data)
+{
+	FuUtilPrivate *priv = (FuUtilPrivate *) user_data;
+	g_debug ("Handling SIGINT");
+	g_cancellable_cancel (priv->cancellable);
+	return FALSE;
+}
+
+static void
+fu_util_private_free (FuUtilPrivate *priv)
+{
+	if (priv->client != NULL)
+		g_object_unref (priv->client);
+	if (priv->soup_session != NULL)
+		g_object_unref (priv->soup_session);
+	g_main_loop_unref (priv->loop);
+	g_object_unref (priv->cancellable);
+	g_option_context_free (priv->context);
+	g_free (priv);
+}
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunused-function"
+G_DEFINE_AUTOPTR_CLEANUP_FUNC(FuUtilPrivate, fu_util_private_free)
+#pragma clang diagnostic pop
+
+int
+main (int argc, char *argv[])
+{
+	gboolean ret;
+	gboolean verbose = FALSE;
+	g_autoptr(FuUtilPrivate) priv = g_new0 (FuUtilPrivate, 1);
+	g_autoptr(GError) error = NULL;
+	g_autoptr(GPtrArray) cmd_array = fu_util_cmd_array_new ();
+	g_autofree gchar *cmd_descriptions = NULL;
+	const GOptionEntry options[] = {
+		{ "verbose", 'v', 0, G_OPTION_ARG_NONE, &verbose,
+			/* TRANSLATORS: command line option */
+			_("Show extra debugging information"), NULL },
+		{ NULL}
+	};
+
+	setlocale (LC_ALL, "");
+
+	bindtextdomain (GETTEXT_PACKAGE, LOCALEDIR);
+	bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
+	textdomain (GETTEXT_PACKAGE);
+
+	/* ensure D-Bus errors are registered */
+	fwupd_error_quark ();
+
+	/* create helper object */
+	priv->loop = g_main_loop_new (NULL, FALSE);
+
+	/* add commands */
+	fu_util_cmd_array_add (cmd_array,
+			       "sync", NULL,
+			       /* TRANSLATORS: command description */
+			       _("Sync the current system status with the server"),
+			       fu_util_sync);
+	fu_util_cmd_array_add (cmd_array,
+			       "register", NULL,
+			       /* TRANSLATORS: command description */
+			       _("Register with a remote server"),
+			       fu_util_register);
+	fu_util_cmd_array_add (cmd_array,
+			       "unregister", NULL,
+			       /* TRANSLATORS: command description */
+			       _("Unregister with a remote server"),
+			       fu_util_unregister);
+	fu_util_cmd_array_add (cmd_array,
+			       "enable", "SERVER",
+			       /* TRANSLATORS: command description */
+			       _("Enable by configuring a server to use"),
+			       fu_util_enable);
+	fu_util_cmd_array_add (cmd_array,
+			       "disable", NULL,
+			       /* TRANSLATORS: command description */
+			       _("Disable by clearing the configured server"),
+			       fu_util_disable);
```

Comment:
What do you think about combining enable/register and disable/unregister?

Reply:
I actually had that originally. Because the registration can fail, it was actually really useful to be able to do enable once, and then be able to do register multiple times. If that's too weird, I can merge them back again.

Reply:
I do think I would prefer to see them merged back.  If register fails one time, just re-register (and enable while you do that).